src/test/java/support/fakes/EndpointFailureDescriptor.java

```diff
@@ -0,0 +1,46 @@
+package support.fakes;
+
+import java.util.Date;
+
+public class EndpointFailureDescriptor {
```

Comment:
What is an endpoint failure descriptor?

Reply:
I have introduced new endpoint for fakes, it is used to emulate resource failures during some period. This feature is used to test that relationships fetch failures are forwarded to client.